Lib/tokenize.py

```diff
@@ -194,7 +194,7 @@ def escape_brackets(self, token):
                 else:
                     characters.append(character)
             if character == "{":
-                if characters[-2:] != ["\\", "N"]:
+                if characters[-2:] != ["\\", "N"] or characters[-3:] == ["\\", "\\", "N"]:
```

Comment:
What if it is `\\\N`? Or `\\\\N`, `\\\\\N`?

Reply:
With this:
```
from io import StringIO
from tokenize import untokenize, generate_tokens
codes = [
        "f'\\\\N{SNAKE}'",
        "f'\\\N{SNAKE}'",
        "f'\\\\N{SNAKE}'",
        "f'\\\\\N{SNAKE}'",
]

for code in codes:
    print(code)
    print(untokenize(generate_tokens(StringIO(code).readline)))
    print(code == untokenize(generate_tokens(StringIO(code).readline)))
```

I get:

```
f'\\N{SNAKE}'
f'\\N{SNAKE}'
True
f'\🐍'
f'\🐍'
True
f'\\N{SNAKE}'
f'\\N{SNAKE}'
True
f'\\🐍'
f'\\🐍'
True
```

Reply:
What specific cases you are thinking about?

Reply:
In `...\\\N{...` we can't determine the semantic by analyzing the limited number of the prefix characters.

Reply:
Yeah but unless I am missing something the current behaviour roundtrips correctly `"f'\\\N{SNAKE}'"` (see above) and `"f'\\\N{{SNAKE}}'"` is illegal (` unknown Unicode character name`) so that's fine. I am missing something?

Reply:
You used non-raw strings in the above example (it reduces the number of backslashes in the input by 2 times). You have maximum 2 backslashes, and this already handled by the code. What if use raw strings or increase the number of backslashes?

Reply:
> What if use raw strings or increase the number of backslashes?

Ah, I understand now. Let me check and will get back to you